From 1b21115318d9620ac7770f1051a536c8a8f3139f Mon Sep 17 00:00:00 2001
Message-Id: <1b21115318d9620ac7770f1051a536c8a8f3139f.1615813233.git.kevin@kevinlocke.name>
From: Kevin Locke <kevin@kevinlocke.name>
Date: Sat, 13 Mar 2021 10:18:37 -0700
Subject: [PATCH v2] dh_bash-completion: Tighten is_filelist matching

The regular expressions in is_filelist which matches "well-known idioms
on bash scripts" currently matches the path to the bash-completion
script in the nitrokey-app package:

    'data/bash-autocomplete/nitrokey-app' =~ /\s*complete.*-[A-Za-z].*/

Avoid this by ensuring the is only matched when following a line break
or character which can be used to chain shell commands.

Signed-off-by: Kevin Locke <kevin@kevinlocke.name>
---
 debian/extra/debhelper/dh_bash-completion | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Changes in v2:
* Match & and | in addition to ; as command separators.
  This is important for `... && complete`, which is a common idiom.

diff --git a/debian/extra/debhelper/dh_bash-completion b/debian/extra/debhelper/dh_bash-completion
index d1d9bf2e..7999151c 100755
--- a/debian/extra/debhelper/dh_bash-completion
+++ b/debian/extra/debhelper/dh_bash-completion
@@ -75,7 +75,7 @@ sub is_filelist {
 		#
 		# - If we see an "if...then" construction in the file.  We
 		#   take into account multi-line statements.
-		if (/\s*complete.*-[A-Za-z].*/
+		if (/(^|[|&;])\s*complete.*-[A-Za-z].*/
 			|| /\$\(.*\)/
 			|| /\s*compgen.*-[A-Za-z].*/
 			|| /\s*if.*;.*then/s) {
-- 
2.30.2

